### PR TITLE
[PATCH v3] api: packet: specify which packet flags cannot be set simultaneously

### PR DESCRIPTION
**Pull request by MatiasElo**

### CODE REVIEW

File: test/validation/api/packet/packet.c [hunk 1]
```diff
@@ -339,23 +340,37 @@ static void packet_set_inflags(odp_packet_t pkt, int val)
...
-	odp_packet_has_ipv6_set(pkt, val);
 	odp_packet_has_ip_bcast_set(pkt, val);
-	odp_packet_has_ip_mcast_set(pkt, val);
 	odp_packet_has_ipfrag_set(pkt, val);
 	odp_packet_has_ipopt_set(pkt, val);
 	odp_packet_has_ipsec_set(pkt, val);
 	odp_packet_has_udp_set(pkt, val);
-	odp_packet_has_tcp_set(pkt, val);
-	odp_packet_has_sctp_set(pkt, val);
-	odp_packet_has_icmp_set(pkt, val);
```
Comment (JereLeppanen) [inline test/validation/api/packet/packet.c:354]: With this change, we're not setting these flags to 1 anywhere in validation. Should we?

Comment (MatiasElo) [reply test/validation/api/packet/packet.c:354]: They are still tested in `packet_test_in_flags()`with some macro trickery.

Comment (JereLeppanen) [reply test/validation/api/packet/packet.c:354]: In that case this should be fine.

**Review by JereLeppanen — APPROVED** [review]

**Review by psavol — APPROVED** [review]

**Review by kkirankk — APPROVED** [review]

**Comment by kkirankk** [comment]
Reviewed-by: Kiran Kumar K <kirankumark@marvell.com>

**Pull request merged**